lib/site.taxonomy.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Module dependencies.
+ */
+import Term from './site.taxonomy.term';
+
+/**
+ * Taxonomy class
+ */
+class Taxonomy {
+	/**
+	 * Create a Taxonomy instance
+	 *
+	 * @param {String} siteId - site id
+	 * @param {String} taxonomy - taxonomy type
+	 * @param {WPCOM} wpcom - wpcom instance
+	 * @return {Null} null
+	 */
+	constructor( siteId, taxonomy, wpcom ) {
```

Comment:
Generally we are passing first the id/slug of the entity that we are working and then passing the other ones _going up_ to get the wpcom instance. For instance in site.comment.like:

```es6
/**
 * CommentLike methods
 *
 * @param {String} cid - comment id
 * @param {Number|String} sid - site id
 * @param {WPCOM} wpcom - wpcom instance
 * @return {Null} null
 */
function CommentLike( cid, sid, wpcom ) {
  // ...
};
```

On other hand it's time to write a guide about name convention (`SiteId` vs `sid` vs `siteID`), code style, etc. but not in this PR :-D